cloud-sql/mysql/tink/main.py

```diff
@@ -0,0 +1,237 @@
+    host_args = db_host.split(":")
+    db_hostname, db_port = host_args[0], int(host_args[1])
+
+    pool = sqlalchemy.create_engine(
+        # Equivalent URL:
+        # mysql+pymysql://<db_user>:<db_pass>@<db_host>:<db_port>/<db_name>
+        sqlalchemy.engine.url.URL(
+            drivername="mysql+pymysql",
+            username=db_user,  # e.g. "my-database-user"
+            password=db_pass,  # e.g. "my-database-password"
+            host=db_hostname,  # e.g. "127.0.0.1"
+            port=db_port,  # e.g. 3306
+            database=db_name,  # e.g. "my-database-name"
+        ),
+        **db_config
+    )
+
+    return pool
+
+
+def init_unix_connection_engine(db_config):
+    # Remember - storing secrets in plaintext is potentially unsafe. Consider using
+    # something like https://cloud.google.com/secret-manager/docs/overview to help keep
+    # secrets secret.
+    db_user = os.environ["DB_USER"]
+    db_pass = os.environ["DB_PASS"]
+    db_name = os.environ["DB_NAME"]
+    db_socket_dir = os.environ.get("DB_SOCKET_DIR", "/cloudsql")
+    cloud_sql_connection_name = os.environ["CLOUD_SQL_CONNECTION_NAME"]
+
+    pool = sqlalchemy.create_engine(
+        # Equivalent URL:
+        # mysql+pymysql://<db_user>:<db_pass>@/<db_name>?unix_socket=<socket_path>/<cloud_sql_instance_name>
+        sqlalchemy.engine.url.URL(
+            drivername="mysql+pymysql",
+            username=db_user,  # e.g. "my-database-user"
+            password=db_pass,  # e.g. "my-database-password"
+            database=db_name,  # e.g. "my-database-name"
+            query={
+                "unix_socket": "{}/{}".format(
+                    db_socket_dir,  # e.g. "/cloudsql"
+                    cloud_sql_connection_name)  # i.e "<PROJECT-NAME>:<INSTANCE-REGION>:<INSTANCE-NAME>"
+            }
+        ),
+        **db_config
+    )
+
+    return pool
+
+
+def init_db():
+    db = init_connection_engine()
+
+    # Create tables (if they don't already exist)
+    with db.connect() as conn:
+        conn.execute(
+            "CREATE TABLE IF NOT EXISTS votes "
+            "( vote_id SERIAL NOT NULL, time_cast timestamp NOT NULL, "
+            "candidate CHAR(6) NOT NULL, voter_email VARBINARY(255), "
+            "PRIMARY KEY (vote_id) );"
+        )
+    return db
+
+
+
+# [START cloud_sql_mysql_cse_query]
+def list_votes(db, eaead):
+    import sqlalchemy
+    
+    with db.connect() as conn:
+        # Execute the query and fetch all results
+        recent_votes = conn.execute(
+            "SELECT candidate, time_cast, voter_email FROM votes "
+            "ORDER BY time_cast DESC LIMIT 5"
+        ).fetchall()
+
+        print("Team\tEmail\tTime Cast")
+        
+        for row in recent_votes:
+            team = row[0]
+            email = eaead.decrypt(row[2], team.encode()).decode()
+            time_cast = row[1]
+
+            # Print recent votes
+            print("{}\t{}\t{}".format(team, email, time_cast))
```

Comment:
nit: prefer f-strings for formatting 
```suggestion
            print(f"{team}\t{email}\t{time_cast}")
```

Reply:
Done